AK/BigIntBase.h

```diff
@@ -0,0 +1,566 @@
+/*
+ * Copyright (c) 2023, Dan Klishch <danilklishch@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/BuiltinWrappers.h>
+#include <AK/Checked.h>
+#include <AK/Span.h>
+#include <AK/StdLibExtras.h>
+#include <AK/Types.h>
+
+namespace AK {
+
+// Ideally, we want to store data in the native processor's words. However, for some algorithms,
+// particularly multiplication, we require double of the amount of the native word size.
+#if defined(__SIZEOF_INT128__) && defined(AK_ARCH_64_BIT)
+using NativeWord = u64;
+using DoubleWord = unsigned __int128;
+using SignedDoubleWord = __int128;
+#else
+using NativeWord = u32;
+using DoubleWord = u64;
+using SignedDoubleWord = i64;
+#endif
+
+template<bool sign>
+using ConditionallySignedDoubleWord = Conditional<sign, SignedDoubleWord, DoubleWord>;
+
+template<typename T>
+concept BuiltInUFixedInt = OneOf<T, bool, u8, u16, u32, u64, unsigned long, unsigned long long, DoubleWord>;
+
+constexpr size_t word_size = sizeof(NativeWord) * 8;
```

Comment:
it seems that `sizeof(X) * 8` is used here and there—it could be clearer to extract it to `bitwidth(X)`…? 🤔